test/worker_test.py

```diff
@@ -990,9 +968,12 @@ def run(self):
 class AssistantTest(unittest.TestCase):
     def setUp(self):
         self.sch = CentralPlannerScheduler(retry_delay=100, remove_delay=1000, worker_disconnect_delay=10)
-        self.w = Worker(scheduler=self.sch, worker_id='X')
+        self.w = Worker(scheduler=self.sch, worker_id='X').__enter__()
```

Comment:
I thought so too. Somebody asked the same question and unfortunately there doesn't seem to be a better way really. http://stackoverflow.com/questions/8416208/in-python-is-there-a-good-idiom-for-using-context-managers-in-setup-teardown

Note that there still are multiple tests where I do use it as a context manager.